docs/content/docs/build/container-states-events.md

```diff
@@ -0,0 +1,336 @@
+
+<!-- TO MODIFY THIS DIAGRAM, SEE INSTRUCTIONS AT THE BOTTOM OF THIS FILE. -->
+![A state diagram of the two possible Synchronization states](./images/SynchronizationStates.png)
+
+{{< callout note >}}
+
+The dotted arrow represents a boolean guard condition. It means that the container cannot transition out of the **dirty** state unless the container is published and connected.
+
+{{< /callout >}}
+
+- **dirty**: A container is in **dirty** state on a client in two situations:
+
+    - It is unpublished. 
+    - Changes have been made to the data on the client but not yet sent to the Fluid service, or sent but not yet acknowledged by the service.
+
+- **saved**: A container is in **saved** state on a client when the container has been published and the service has acknowledged all data changes made on the client. 
+
+When a new change is made to the data, the container moves to **dirty** state. When all pending changes are acknowledged, it transitions to the **saved** state. 
+
+Note that a container in the **saved** state is not necessarily perfectly synchronized with the service. 
+There may be changes made on other clients that have been saved to the service but have not yet been relayed to this client. 
+Situations like this would normally last only fractions of a second. 
+But if the client is disconnected while in **saved** state, it remains **saved**, but unsynchronized, until it reconnects. See [Connection status states](#connection-status-states) for more about connection.
+
+Users can work with the container's data regardless of whether it is in **dirty** or **saved** state. 
+But there are scenarios in which your code must be aware of the container's state. For this reason, the `container` object has a boolean `isDirty` property to specify its state. 
+The `container` object also supports *dirty* and *saved* events, so you can handle the transitions between states.
+The container emits the *saved* event to notify the caller that all the local changes have been acknowledged by the service.
+
+```typescript {linenos=inline}
+container.on("saved", () => {
+    // All pending edits have been saved.
+});
+```
+
+The container emits the *dirty* event to notify the caller that there are local changes that have not been acknowledged by the service yet.
+
+```typescript {linenos=inline}
+container.on("dirty", () => {
+    // The container has pending changes that need to be acknowledged by the service.
+});
+```
+
+You should always check the `isDirty` flag before [disposing](#disposed) the container or [disconnecting](#disconnected) from the service.
+If you dispose or disconnect the container while `isDirty === true`, you may lose operations that have not yet been acknowledged by the service.
```

Comment:
```suggestion
If you dispose or disconnect the container while `isDirty === true`, you may lose operations that have not yet been sent to (or acknowledged by) the service.
```

An interesting scenario here is that an operation might have been sent to the service, the container disconnects/disposes, but the service still manages to sequences that operation thus making it durable even if the client that sent it didn't get the acknowledgement. I think this is possible in principle. The main scenario where operations can be lost is when they have not been sent to the service at all.